feature-libs/product-configurator/rulebased/components/form/configurator-form.component.html

```diff
@@ -27,13 +27,13 @@
           [groupType]="currentGroup.groupType"
         ></cx-configurator-attribute-header>
 
-        <!-- Single Selection Radio Button -->
-        <cx-configurator-attribute-radio-button
-          *ngIf="attribute.uiType === uiType.RADIOBUTTON"
-          [attribute]="attribute"
-          [ownerKey]="configuration.owner.key"
-          (selectionChange)="updateConfiguration($event)"
-        ></cx-configurator-attribute-radio-button>
+        <div
+          [cxInnerComponentsHost]="{
+            attribute: attribute,
+            configuration: configuration,
+            group: currentGroup
+          }"
+        ></div>
```

Comment:
To my understanding and vision, the `configurator-form.component.html` should contain almost only this element with `cxInnerComponentsHost` directive (+/- surrounding things, if any, e.g. for styling)

Ideally, I'd prefer the context-provider mechanism to be a generic and independent mechanism. Independent from specific extensibility mechanisms (outlet, innerComponents, whatever....).
In particular, I'd not pass the context via the `@input` of specific extenisbility directive, like here.

I described the idea roughly in the ticket: https://github.com/SAP/spartacus/issues/14332, however it needs some work to PoC it, then overcome possible obstacles found and finally polish the API. This doesn't sound like a low hanging fruit. However a very juicy one we definitely want to have soon. 
It's a matter of other competing priorities and team capacity, when we start doing it.

Reply:
> I guess it was also your intention and you took only one component by example just for PoC purpose. But let me clarify so we're on the same page: To my understanding and vision, the configurator-form.component.html should contain almost only this element with cxInnerComponentsHost directive (+/- surrounding things, if any, e.g. for styling). All concrete attribute components should be listed in the inner components configuration.

Yes, completely agree. I only added the single attribute component in order to try things out